examples/serialization.py

```diff
@@ -0,0 +1,122 @@
+# Copyright © 2022 United States Government as represented by the Administrator of the
+# National Aeronautics and Space Administration.  All Rights Reserved.
+
+"""
+Example of serializing and de-serializing a surrogate model using JSON and pickling methods
+"""
+
+from prog_models.models import BatteryElectroChemEOD as Battery
+from prog_models.data_models import DMDModel
+
+import matplotlib.pyplot as plt
+import pickle
+
+def run_example():  
+    ## Step 1: Create a model object
+    batt = Battery()
+
+    ## Step 2: Define future loading functions for training data 
+    def future_loading_1(t, x=None):
+        # Variable (piece-wise) future loading scheme 
+        if (t < 500):
+            i = 3
+        elif (t < 1000):
+            i = 2
+        elif (t < 1500):
+            i = 0.5
+        else:
+            i = 4.5
+        return batt.InputContainer({'i': i})
+    
+    def future_loading_2(t, x=None):
+        # Variable (piece-wise) future loading scheme 
+        if (t < 300):
+            i = 2
+        elif (t < 800):
+            i = 3.5
+        elif (t < 1300):
+            i = 4
+        elif (t < 1600):
+            i = 1.5
+        else:
+            i = 5
+        return batt.InputContainer({'i': i})
+    
+    load_functions = [future_loading_1, future_loading_2]
+
+    ## Step 3: Create surrogate model 
+    options_surrogate = {
+        'save_freq': 1, # For DMD, this value is the time step for which the surrogate model is generated
+        'dt': 0.1, # For DMD, this value is the time step of the training data
+        'trim_data_to': 0.7 # Value between 0 and 1 that determines the fraction of data resulting from simulate_to_threshold that is used to train DMD surrogate model
+    }
+
+    # Set noise in Prognostics Model, default for surrogate model is also this value. Set to 0 to illustrate match between original surrogate and serialized version
+    batt.parameters['process_noise'] = 0 
+
+    # Generate surrogate model  
+    surrogate_orig = batt.generate_surrogate(load_functions,**options_surrogate)
+
+    ### Step 4: serialize model for future use 
+    # METHOD 1: Serialize with JSON 
+    save_surrogate = surrogate_orig.to_json()
+
+    # DMDModel can be called directly with serialized result
+    surrogate_serial_1 = DMDModel.from_json(save_surrogate)
+
+    # Serialized result can also be saved to a text file and uploaded later using the following code:
+    txtFile = open("surrogate_model_save_json.txt", "w")
+    txtFile.write(save_surrogate)
+    txtFile.close() 
+
+    with open('surrogate_model_save_json.txt') as infile: 
+        load_surrogate_json = infile.read()
+
+    surrogate_serial_2 = DMDModel.from_json(load_surrogate_json)
+
+    # METHOD 2: Serialize by pickling
+    pickle.dump(surrogate_orig, open('surrogate_model_save_pkl.pkl','wb'))
+    surrogate_pkl = pickle.load(open('surrogate_model_save_pkl.pkl','rb'))
+
+    ## Step 5: Simulate to threshold and compare results
```

Comment:
```suggestion
    # Note: Model serialization has a lot of purposes, like saving a specific model to a file to be loaded later or sending a model to another machine over a network connection. 
    
    ## Step 5: Simulate to threshold and compare results
```

Reply:
I like this comment a lot. Is it okay to move it upwards a bit in the example, to where we first use serialization? See change in updated PR